core/src/main/kotlin/riven/core/configuration/websocket/WebSocketSecurityInterceptor.kt

```
@@ -0,0 +1,115 @@
+     * On success, sets the authenticated principal on the STOMP session so subsequent
+     * frames (SUBSCRIBE, SEND) can access the user's identity and authorities.
+     */
+    private fun handleConnect(accessor: StompHeaderAccessor) {
+        val token = extractBearerToken(accessor)
+            ?: throw AuthenticationCredentialsNotFoundException("Missing Authorization header on CONNECT")
+
+        try {
+            val jwt = jwtDecoder.decode(token)
+            val authentication = tokenConverter.convert(jwt)
+            accessor.user = authentication
+        } catch (e: JwtException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)